modules/platforms/cpp/thin-client/include/ignite/thin/cache/event/cache_entry_event_listener.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file
+ * Declares ignite::thin::cache::event::CacheEntryEventListener class.
+ */
+
+#ifndef _IGNITE_THIN_CACHE_EVENT_CACHE_ENTRY_EVENT_LISTENER
+#define _IGNITE_THIN_CACHE_EVENT_CACHE_ENTRY_EVENT_LISTENER
+
+#include <stdint.h>
+
+#include <ignite/thin/cache/event/cache_entry_event.h>
+
+namespace ignite
+{
+    namespace thin
+    {
+        namespace cache
+        {
+            namespace event
+            {
+                /**
+                 * Cache entry event listener.
+                 */
+                template<typename K, typename V>
+                class CacheEntryEventListener
```

Comment:
Can we provide `OnDisconnected` here as well, so the user knows that continuous query has stopped due to connection loss? Or is there some other API to deal with this situation?

Reply:
That's a really good idea, I've somehow missed this part. Need to add a test too.